src/search/view/SearchListView.ts

```diff
@@ -399,6 +398,23 @@ export class SearchListView implements Component {
 		}
 	}
 
+	toggleUnreadStatus(): void {
+		let selected = this.getSelectedEntities()
+
+		if (selected.length > 0) {
+			if (isSameTypeRef(selected[0].entry._type, MailTypeRef)) {
+				let selectedMails = selected.map((m) => m.entry as any as Mail)
```

Comment:
could we use a type assertion function here that asserts that the array is an array of mail entities to convince the compiler? `as any as T` is scary :)

Reply:
`as unknown as T` would be a tiny bit better but generally it's safe imo